### PR TITLE
Fix date_format function in fe

### LINKED ISSUE #683: date_format function incorrect
**Describe the bug**
mysql> select date_format("2019-03-04 22:23:00", '%W %w');
+---------------------------------------------+
| date_format('2019-03-04 22:23:00', '%W %w') |
+---------------------------------------------+
| %2 %10                                      |
+---------------------------------------------+
1 row in set (0.08 sec)

**To Reproduce**
mysql> select date_format("2019-03-04 22:23:00", '%W %w');
+---------------------------------------------+
| date_format('2019-03-04 22:23:00', '%W %w') |
+---------------------------------------------+
| %2 %10                                      |
+---------------------------------------------+
1 row in set (0.08 sec)

**Expected behavior**
mysql> select date_format("2019-03-04 22:23:00", '%W %w');
+---------------------------------------------+
| date_format('2019-03-04 22:23:00', '%W %w') |
+---------------------------------------------+
| Monday 1                                    |
+---------------------------------------------+
1 row in set (0.04 sec)

### PR DESCRIPTION
**Pull request by HangyuanLiu**
Add a new date_format function in the fe module to facilitate pruning in the fe analysis phase
fix : #683

### CODE REVIEW

File: fe/src/main/java/org/apache/doris/rewrite/FEFunctions.java [hunk 1]
```diff
@@ -121,6 +125,210 @@ private static long getTime(LiteralExpr expr) throws AnalysisException {
...
+            }
+            else if (character == '%') {
+                escaped = true;
+            }
+            else {
+                formatterBuilder.appendLiteral(character);
+            }
+        }
+        DateTimeFormatter formatter = formatterBuilder.toFormatter();
+        return formatter.withLocale(Locale.US).print(date.getTime());
```
Comment (morningman) [inline fe/src/main/java/org/apache/doris/rewrite/FEFunctions.java]: Why Locale.US?

**Review by morningman — COMMENTED** [review]
Can you add some unit test for this function?

File: fe/src/main/java/org/apache/doris/rewrite/FEFunctions.java (hunk 1 above)
Comment (HangyuanLiu) [reply fe/src/main/java/org/apache/doris/rewrite/FEFunctions.java]: Without this statement, AM will be changed to "上午"

File: fe/src/main/java/org/apache/doris/rewrite/FEFunctions.java [hunk 2]
```diff
@@ -121,6 +125,210 @@ private static long getTime(LiteralExpr expr) throws AnalysisException {
...
+        calendar.set(year, Calendar.JANUARY,1);
+        int firstDay = 1;
+        calendar.set(Calendar.DAY_OF_MONTH, firstDay);
+        while (calendar.get(Calendar.DAY_OF_WEEK) != firstWeekDay) {
+            calendar.set(Calendar.DAY_OF_MONTH, ++firstDay);
+        }
+        return firstDay;
+    }
+
+    private  static String dateFormatUtils(Date date,  String pattern) {
```
Comment (imay) [inline fe/src/main/java/org/apache/doris/rewrite/FEFunctions.java]:
```suggestion
    private  static String dateFormat(Date date,  String pattern) {
```

**Review by imay — APPROVED** [review]
LGTM

Good job! Thanks for your contribution.

**Pull request merged**